### PR TITLE
Clean up .gitignore, dropping /translate/.gitignore as unnecessary

### PR DESCRIPTION
**Pull request by eemeli**
Reorganise the rules into sections, and sort alphabetically: prefixes, suffixes, matches anywhere, and matches at root. Adds `.*` to catch all hidden files by default (they can still be added with `--force`). With that, `/translate/.gitignore` becomes unnecessary.

Filing this, as I want to add a `.python-version` file in my local pontoon dir to have pyenv automatically activate the right virtualenv when `cd`'ing into the project.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,34 +1,27 @@
...
+*~
+
+build.py
+build/
 coverage/
-*.log
+dist/
+node_modules/
+tmp/
+venv/
```
Comment (flodolo) [inline .gitignore:18]: We lose `.venv` with this?

Comment (eemeli) [reply .gitignore:18]: No, that's covered by `.*`.

File: .gitignore [hunk 2]
```diff
@@ -1,34 +1,27 @@
...
 
-# Docker
-.server-build
-docker/config/server.env
-docker/config/server.env-e
+/docker/config/server.env
+/docker/config/server.env-e
```
Comment (mathjazz) [inline .gitignore:21]: Is that still a thing?

**Review by mathjazz — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (eemeli) [reply .gitignore:21]: Good point, I'll drop the rule. Note that you'll need to manually remove the file if you still have it; the one in my local clone was last touched in February 2022.
```suggestion
```

**Pull request merged**